### PR TITLE
Custom intcmp values that are valid during a flash sale

### PR DESCRIPTION
**Pull request by lmath**
## Why are you doing this?
During a flash sale, it is useful to be able to set a custom value for the internal campaign code to more easily track the promotion in GA.

This adds that capability.

This does mean that there is a map of intcmp values that are valid during the flash sale that needs to be manually updated whenever we set up another flash sale if we are going to use a specific intcmp value for tracking.

The particular intcmp values specified in this PR relate to this flash sale: https://github.com/guardian/support-frontend/pull/850

[**Trello Card**](https://trello.com/c/3wVpGbeA/1751-voucher-home-delivery-flash-sale-updates)

## Changes

* Particular products during a flash sale can now have a specific intcmp value. If there is one specified, it is used if the flash sale is live, otherwise the default value is used.

### CODE REVIEW

File: assets/helpers/externalLinks.js [hunk 1]
```diff
@@ -152,13 +176,13 @@ function getSubsLinks(
     return buildSubsUrls(
       countryGroupId,
       customPromos[campaign],
-      intCmp,
+      defaultIntcmps,
```
Comment (Ap0c) [inline assets/helpers/externalLinks.js]: Am I right in thinking this makes the `intCmp` param to this function effectively unused? As a result, will this always overwrite any `INTCMP` that the user has when they arrive on support (i.e. from dotcom)?

File: assets/helpers/externalLinks.js [hunk 2]
```diff
@@ -111,25 +122,38 @@ function getMemLink(product: MemProduct, intCmp: ?string): string {
...
+  intCmp: Intcmps,
   otherQueryParams: Array<[string, string]>,
   referrerAcquisitionData: ReferrerAcquisitionData,
 ): SubsUrls {
 
   const countryId = countryGroups[countryGroupId].supportInternationalisationId;
   const params = new URLSearchParams();
-  params.append('INTCMP', intCmp || defaultIntCmp);
   otherQueryParams.forEach(p => params.append(p[0], p[1]));
   params.append('acquisitionData', JSON.stringify(referrerAcquisitionData));
```
Comment (Ap0c) [inline assets/helpers/externalLinks.js]: Are these three lines needed now that `buildParamString` exists?

File: assets/helpers/externalLinks.js (hunk 1 above)
Comment (lmath) [reply assets/helpers/externalLinks.js]: Well spotted. I'm on it.

**Comment by JustinPinner** [comment]
👍 :shipit: 🚢

**Pull request merged**